bioconda_utils/recipe.py

```diff
@@ -206,6 +210,32 @@ def load_from_string(self, data) -> "Recipe":
         self.render()
         return self
 
+    def read_conda_build_config(self):
+        # Cache contents of conda_build_config.yaml for conda_render.
+        path = Path(self.dir, 'conda_build_config.yaml')
+        if path.is_file():
+            self.conda_build_config = path.read_text()
+        else:
+            self.conda_build_config = ''
+
+    def read_build_scripts(self):
+        # Cache contents of build scripts for conda_render since conda-build
+        # inspects build scripts for used variant variables.
```

Comment:
This keeps the build scripts (and `conda_build_config.yaml` if present) in memory, increasing the memory usage of `bioconda-utils update-pinning` even more. I haven't checked by how much, though. `bioconda-utils update-pinning` in general chews through a good amount of memory. Just beware that you should run `bioconda-utils update-pinning` on a machine with more than a couple of gigs of RAM ;).